src/config.rs

```diff
@@ -1,63 +1,69 @@
-use std::{error::Error, fs, path::Path};
-
-use crate::{app::TaskStore, task::Task, theme::Theme};
-
-// FIX: Proper handling, data should not be stored in the config file and needes testing
-// Add a swp file.
-pub fn get_data() -> Result<(Theme, TaskStore), Box<dyn Error>> {
-    match dirs::home_dir() {
-        Some(home_dir) => {
-            let config_path = Path::new(&home_dir).join(".config/dotodo/config.yml");
-            let data_path = Path::new(&home_dir).join(".config/dotodo/data.json");
-
-            let theme = if !config_path.exists() {
-                Theme::default()
-            } else {
-                let config_contents = fs::read_to_string(&config_path);
-                match config_contents {
-                    Ok(file) => serde_yaml::from_str::<Theme>(&file)?,
-                    Err(_) => {
-                        let theme = Theme::default();
-                        fs::write(&config_path, serde_yaml::to_string(&theme)?)?;
-                        theme
-                    }
-                }
-            };
-
-            let task_store = if !data_path.exists() {
-                TaskStore::default()
-            } else {
-                let data_contents = fs::read_to_string(&data_path);
-                match data_contents {
-                    Ok(file) => serde_json::from_str::<TaskStore>(&file)?,
-                    Err(_) => {
-                        let tasks: Vec<Task> = vec![];
-                        fs::write(&data_path, serde_json::to_string(&tasks)?)?;
-                        TaskStore::default()
-                    }
+use std::{error::Error, fs};
+
+use crate::{app::TaskStore, theme::Theme};
+
+const DIR: &str = "dotodo";
+
+const CONFIG_FILE: &str = "config.yml";
+const DATA_FILE: &str = "data.json";
+
+pub fn get_data() -> (Theme, TaskStore) {
+    let config_local_dir = dirs::config_local_dir();
+    let data_local_dir = dirs::data_local_dir();
+
+    let mut theme = Theme::default();
+
+    if let Some(dir) = config_local_dir {
+        let config_path = dir.join(DIR).join(CONFIG_FILE);
+
+        if config_path.exists() {
+            if let Ok(config) = fs::read_to_string(&config_path) {
+                if let Ok(theme_config) = serde_yaml::from_str::<Theme>(&config) {
```

Comment:
This should probably be handled with a corruption message and if they want to quit without saving, as to avoid accidentally destroying data.

Reply:
Will a command-line prompt suffice for now? Smth like
```
Tasks data seems to be corrupted. If you continue, it will be overwritten.
Continue (y/n)? y
```

As for error messages, see my answer to next review.

Reply:
> Will a command-line prompt suffice for now?
Sounds good